apis/v1/httproute_types.go

```diff
@@ -347,7 +347,7 @@ type HTTPRouteTimeouts struct {
 
 	// BackendRequest specifies a timeout for an individual request from the gateway
 	// to a backend. This covers the time from when the request first starts being
-	// sent from the gateway to when the full response has been received from the backend.
+	// sent from the gateway to when the response headers have been received from the backend.
```

Comment:
This seems like a significant change, would appreciate some additional reviews on this.

/cc @youngnick @howardjohn @arkodg @kate-osborn @mlavacca

Reply:
IIUC this breaks Envoy implementations?

Reply:
Isn't it the opposite? Envoy can do "end of response" not "end of response headers"

Reply:
🤦 Looking at the wrong timeout section. Let me edit my comment.

Reply:
@youngnick Ok I see, so I will rework the changes to only fix the diagram, if it is ok for you.

Reply:
Thanks and sorry for the delay here.

Reply:
@youngnick I have updated the PR, nonetheless, I have a question regarding the `timeouts.request` end time, the go doc says:

> Request specifies the maximum duration for a gateway to respond to an HTTP request.

but also

> This timeout is intended to cover as close to the whole request-response

According to this, I think the `timeouts.request` end time is not well placed in diagram, WDYT?

Reply:
That's why it says "as close as" - from my reading of the timeouts available across implementations, there's not enough commonality to be able to say "this timeout covers exactly from here to here and only those things" - whatever we pick, _some_ implementation can't do it.

Reply:
@youngnick Ok, so if it is better to remain vague to match most implementations, could I reflect this tolerance margin on the diagram, as for the `timeouts.request` start time (min and max)?

It makes me wonder why the tolerance margin, regarding the exact covering a timeout has, has not been introduced when defining the `timeouts.backendRequest` timeout as well.
Now that the `timeouts.backendRequest` covering is clarified and aligned in the documentation, it suits Envoy implementation, but not Traefik.

Would you be inclined to introduce a tolerance margin that suits the Traefik implementation?

Reply:
Having a tolerance margin as a field? I need to think about that for a while, sorry.

Reply:
@rtribotte No no, just documenting the tolerance.

Reply:
This comes down to conformance. If we're going to have a margin in conformance tests, then we should definitely document it. That said, I think that the conformance tests only really do human-scale timeouts (waiting 5sec for a 3sec timeout or similar), so the margin may be less relevant.

Reply:
To meet the requirements of the specification, the technical implementation of the timeout should be exactly what is described.
If no tolerance margin is documented, this means that Traefik's implementation does not conform to the specification.

Following what you said earlier:
> That's why it says "as close as" - from my reading of the timeouts available across implementations, there's not enough commonality to be able to say "this timeout covers exactly from here to here and only those things" - whatever we pick, some implementation can't do it.

I suggest we add a statement in the spec to allow a margin for the `timeouts.backendRequest` timeout, allowing for multiple implementations, as done for the `timeouts.request` timeout.

WDYT?

Reply:
Yes, replicating what we have in the `timeouts.request` timeout is fine.

Reply:
@youngnick Ok, I'll adjust the PR for this, thanks!